AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -7,25 +7,69 @@
 //
 
 import UIKit
+import MobileCoreServices
+import AVKit
+import AVFoundation
 
-class DoubleVideoViewController: UIViewController, CellTitled {
+class DoubleVideoViewController: UIViewController, CellTitled, UIImagePickerControllerDelegate,UINavigationControllerDelegate {
+    
+    var movieURL: URL?
+    
     var titleForCell: String = "Double Video"
-
+    
+    let picker = UIImagePickerController()
+    
     @IBOutlet weak var videoContainerTop: UIView!
     @IBOutlet weak var videoContainerBottom: UIView!
     
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        picker.delegate = self
+        
         // don't f with this line
         self.edgesForExtendedLayout = UIRectEdge(rawValue: 0)
     }
-
+    
     override func viewDidLayoutSubviews() {
         // update your layers' frames here
     }
     
     @IBAction func loadVideo(_ sender: UIButton) {
+        picker.sourceType = .photoLibrary
+        picker.mediaTypes = UIImagePickerController.availableMediaTypes(for: .photoLibrary)!
```

Comment:
```swift
 imagePickerController.mediaTypes = [String(kUTTypeMovie)]
 ```

That call would be preferable because it specifies the type we want (videos only) and doesn't require to be force unwrapped. I didn't require video only, but did teach the array of explicit types. You didn't lose any points for the choice to use ```availableMediaTypes(for:)``` but did lose 5% overall on correctness and style for the forced unwrap.